core-rust/state-manager/src/state_manager.rs

```diff
@@ -279,17 +271,17 @@ where
         //========================================================================================
 
         let read_store = self.store.read();
-        let base_accumulator_state = read_store.get_top_accumulator_state();
+        let (base_state_version, base_ledger_hashes) = read_store.get_top_ledger_hashes();
         let epoch_header = read_store
             .get_last_epoch_proof()
             .expect("at least genesis epoch must exist")
             .ledger_header;
         let epoch_identifiers = EpochTransactionIdentifiers::from(&epoch_header);
 
-        if prepare_request.committed_accumulator_state != base_accumulator_state {
+        if prepare_request.committed_ledger_hashes != base_ledger_hashes {
             panic!(
                 "state {:?} from request does not match the current ledger state {:?}",
-                prepare_request.committed_accumulator_state, base_accumulator_state
+                prepare_request.committed_ledger_hashes, base_ledger_hashes
             );
         }
 
```

Comment:
_(note from author)_

The pre-computation of `prepared_proposed_results` will go away in a refactor which will introduce `DuplicateIntentHashDetector` (see https://github.com/radixdlt/babylon-node/pull/474/files#diff-5230880996b770cf628e475ce1143be291e99fd9801f307fd1702a525c351aadR1005).

Reply:
☝️ this happens at https://github.com/radixdlt/babylon-node/pull/480